penguin/web/routes.py

```
@@ -5596,28 +5660,71 @@ async def load_context_file(
         uploads_dir.mkdir(exist_ok=True)
 
-        # Generate a unique filename
-        file_extension = file.filename.split(".")[-1] if "." in file.filename else ""
-        unique_filename = f"{uuid.uuid4()}.{file_extension}"
+        unique_filename = f"{uuid.uuid4()}{extension}"
         file_path = uploads_dir / unique_filename
+        max_upload_bytes = _max_upload_bytes()
+        bytes_written = 0
 
-        # Save the file
         with open(file_path, "wb") as buffer:
```

Ruff (0.15.10)

[warning] Async functions should not open files with blocking methods like `open`

(ASYNC230)